src/csharp/Microsoft.Spark/Sql/Functions.cs

```diff
@@ -4088,7 +4277,7 @@ private static UserDefinedFunction CreateVectorUdf<TResult>(string name, Delegat
                     CommandSerDe.SerializedMode.Row,
                     CommandSerDe.SerializedMode.Row),
                 evalType,
-                UdfUtils.GetReturnType(typeof(TResult)));
+                returnType.Json ?? UdfUtils.GetReturnType(typeof(TResult)));
```

Comment:
Won't evaluating `returnType.Json` throw an exception if `returnType` is null?

Reply:
If `returnType` is null, it will use `UdfUtils.GetReturnType(typeof(TResult)))` instead.

Reply:
Did you test this?  My understanding is that `??` will evaluate `returnType.Json` which will throw `NullReferenceException` if `returnType` is null.  But I might be wrong.

Reply:
Yes, tests will pass when `returnType.Json` is null. `??` operator will return left-hand value which is `returnType.Json` here if it's not null, otherwise will evaluate and return right-hand value. Please refer to [this](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator).

Reply:
I mean when `returnType` is null, not `returnType.Json`.